esmvaltool/diag_scripts/clouds/clouds.ncl

```diff
@@ -197,21 +248,111 @@ begin
   ; # get data and average time               #
   ; ###########################################
 
-  maps = new((/dim_MOD, 4/), graphic)
-  maps_d = new((/dim_MOD, 4/), graphic)
+  ; ---------------------------------------------------------
+  ; if requested, calculate multi-observational mean and standard deviation
 
-  ind_all_sorted = ispan(0, dim_MOD - 1, 1)  ; create array
+  if (flag_multiobs_unc) then
+    nobs = dimsizes(idxobs)
+
+    ; step 1: calculate multi-obs mean
+
+    do i = 0, nobs - 1
+      A0 = read_data(info0[idxobs(i)])
+
+      ; calculate time average
+      mean = time_operations(A0, -1, -1, "average", "annualclim", True)
+      delete(A0)
+
+      ; if requested, extract geographical region
+      if (isatt(diag_script_info, "region")) then
+        region = diag_script_info@region
+        mean := area_operations(mean, region(0), region(1), region(2), \
+                                region(3), "extract", False)
+      end if
+      if (i .eq. 0) then
+        dims = dimsizes(mean)
+        newdims = new(dimsizes(dims) + 1, integer)
+        newdims(0) = nobs
+        newdims(1:dimsizes(newdims) - 1) = dims
+        ref_tmp = new(newdims, float)
+        delete(dims)
+      end if
+      ref_tmp(i, :, :) = mean
+    end do
+    delete(mean)
+    ; note: we are using dim_avg_n_Warp so missing values are ignored
+    ;       when averaging
+    ref_avg = dim_avg_n_Wrap(ref_tmp, 0)
+    delete(ref_tmp)
+
+    ; step 2: calculate standard deviation of all obs datasets using
+    ;         the multi-obs mean
+
+    sigma2 = new(dimsizes(ref_avg), float)
+    sig_tmp = new(newdims, float)
+    delete(newdims)
+
+    do i = 0, nobs - 1
+      A0 = read_data(info0[idxobs(i)])
+      ; calculate yearly averages
+      ymean = time_operations(A0, -1, -1, "average", "yearly", True)
+      delete(A0)
+      ; if requested, extract geographical region
+      if (isatt(diag_script_info, "region")) then
+        region = diag_script_info@region
+        ymean := area_operations(ymean, region(0), region(1), region(2), \
+                                 region(3), "extract", False)
+      end if
+      dims = dimsizes(ymean)
+      nyears = dims(0)
+      delete(dims)
+      sigma2 = 0.0
+      do j = 0, nyears - 1  ; dimension 0 = time
+        sigma2 = sigma2 + (ref_avg - ymean(j, :, :)) ^ 2
+      end do
+      if (nyears .gt. 1) then
+        sigma2 = sigma2 / tofloat(nyears - 1)
+      end if
+      sig_tmp(i, :, :) = sigma2
+      delete(ymean)
+    end do
+    delete(sigma2)
+    ; note: we are using dim_avg_n_Warp so missing values are ignored
+    ;       when averaging
+    ref_std = sqrt(dim_avg_n(sig_tmp, 0))
```

Comment:
The functions dim_avg_n and dim_avg_n_Wrap both ignore missing values, the only difference is whether the metadata is preserved (dim_avg_n_Wrap) or not (dim_avg_n). I updated the comments in clouds.ncl to avoid confusion.